src/basic/string-util.c

```diff
@@ -1219,7 +1220,12 @@ int string_contains_word(const char *string, const char *separators, const char
                         return r;
                 if (r == 0)
                         return false;
```

Comment:
given our usual logic to initialize all return params if we return >= 0 i think we should set `*ret_word = NULL;` here in the `false` branch